wkcuber/wkcuber/upload.py

```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+from argparse import ArgumentParser
+from typing import Optional
+from os import environ
+
+from webknossos import webknossos_context
+from wkcuber.api.dataset import Dataset
+
+from .utils import add_verbose_flag, setup_logging
+
+
+def create_parser() -> ArgumentParser:
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "source_path", help="Directory containing the source WKW dataset.", type=Path
+    )
+
+    parser.add_argument(
+        "--token",
+        help="Auth token of the user on webKnossos",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--url", help="Base url of the webKnossos instance", default=None
+    )
+
+    parser.add_argument(
+        "--jobs",
+        "-j",
+        default=5,
+        type=int,
+        help="Number of simultaneous upload processes.",
+    )
+
+    add_verbose_flag(parser)
+
+    return parser
+
+
+def upload_dataset(
+    source_path: Path,
+    url: str,
+    token: Optional[str],
+    jobs: int,
+) -> None:
+    with webknossos_context(url=url, token=token):
+        Dataset.open(source_path).upload(jobs)
+
+
+if __name__ == "__main__":
+    args = create_parser().parse_args()
+    setup_logging(args)
+    url = (
+        args.url
+        if args.url is not None
+        else environ.get("WK_URL", "https://webknossos.org")
+    )
+    token = args.token if args.token is not None else environ.get("WK_TOKEN", None)
```

Comment:
This duplicates the logic from https://github.com/scalableminds/webknossos-libs/blob/2922e874641e96b50e27819379966579600d8f07/webknossos/webknossos/client/context.py#L123. I think, we should try to DRY this, as this will probably occur more often in the future.

Reply:
The difference is that the `url` has a default value, whereas `token` needs to be supplied. So I don't think it is a duplicate.

Reply:
Hm, what I meant is that that upload.py:58 basically duplicates the code from context.py:123 and upload.py:60 is a duplicate of context.py:124. You could at least import `DEFAULT_WEBKNOSSOS_URL` instead of using the hardcoded literal in L58 to make it a bit DRYer.

Reply:
Now, I understand what you meant. I re-used the constant :+1: